include/private/http/response_parser.hpp

```diff
@@ -181,9 +181,25 @@ class ResponseParserNg : public NonCopyable, public NonMovable {
     size_t parser_execute(const void *p, size_t n) {
         size_t x =
             http_parser_execute(&parser_, &settings_, (const char *)p, n);
+        // FIX: I initially coded `upgrade` as the following commented out code
+        // does, because I did read [the documentation of http-parser](
+        // https://github.com/nodejs/http-parser#the-special-problem-of-upgrade)
+        // but it seems that is written more with servers in mind.
+        //
+        // For a client, unless `upgrade` is requested by us, the resulting
+        // headers are just informational and, as such, we can continue.
+        //
+        // For reference see: <https://tools.ietf.org/html/rfc7230#section-6.7>.
+        //
+        // This fixes for example the `http://aseansec.org/` URL.
+        //
+        // I am going to keep the code below commented out and this comment
+        // so the choices I made are clear and documented.
+        /*
         if (parser_.upgrade) {
             throw UpgradeError();
         }
+        */
```

Comment:
Maybe then you want to just delete the code in here and move the comment above into the commit message?

Reply:
Yeah, makes sense!

Reply:
Yes I agree with this fix. Actually `Upgrade` is used to "kind of" advertise support for HTTP2 (or spdy) so in those cases the client knows it can use it (like you say "informational"). Maybe in the future we want to also support HTTP2 so maybe it's worth referencing this commit in that ticket to know the point of the code where that logic should/could be hooked.